src/wasm/kotlin/javatests/arcs/UnicodeTest.kt

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2019 Google LLC.
+ *
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ *
+ * Code distributed by Google as part of this project is also subject to an additional IP rights
+ * grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+package sdk.kotlin.javatests.arcs
+
+import arcs.addressable.toAddress
+import arcs.Collection
+import arcs.Handle
+import arcs.Particle
+import arcs.Singleton
+import kotlin.native.internal.ExportForCppRuntime
+import kotlin.native.Retain
+
+class UnicodeTest : Particle() {
+    private val sng = Singleton { UnicodeTest_Sng() }
+    private val col = Collection { UnicodeTest_Col() }
+    private val res = Collection { UnicodeTest_Res() }
+
+    init {
+        registerHandle("sng", sng)
+        registerHandle("col", col)
+        registerHandle("res", res)
+    }
+
+    override fun onHandleUpdate(handle: Handle) {
+        val out = UnicodeTest_Res()
+        if (handle.name == "sng") {
+            val data = (handle as Singleton<*>).get() as UnicodeTest_Sng
+            out.txt = data.txt
+        } else {
+            val data = (handle as Collection<*>).iterator().next() as UnicodeTest_Col
+            out.txt = data.txt
+        }
```

Comment:
This could be simplified to:

```kotlin
out.txt = if (handle.name == "sng") {
    ((handle as Singleton<*>).get() as UnicodeTest_Sng).txt
} else {
    ((handle as Collection<*>).first() as UnicodeTest_Col).txt
}
```

Reply:
Also, if the collection is empty - what would happen?